### ISSUE TITLE
Filter prometheus metrics

### ISSUE THREAD

**Issue by 0123hoang** [fluent/fluent-bit#6001]
When scrape Prometheus metrics, maybe there are too many metrics scraped and I want some filter plugin can filter out unwanted metrics. (the grep plugin filter only work with JSON format)

## **Proposal**
Config value "Include/Exclude" have multi values, or have regex strings.
**Metric filter**
```
[FILTER]
    Name  metrics
    Match node_exporter*
    Include  node_memory*,node_cpu_seconds*
```

**Comment by patrick-stephens** [comment]
I agree, having a configurable option as to what aspect of node_exporter to turn on would be useful particularly from a security or config perspective to prevent mounting things you don't need, etc.

**Comment by endersonmaia** [comment]
I propose renaming this issue with something like "Add support to use filters for metric type inputs".

IMO, instead of proposing a new filter, the current filters should work on metrics the same way they work for logs.

For example, using the `record_modifier` and the `allowlist_key` operator should be enough to achieve the example on the issue.


```
[FILTER]
  name            record_modifier
  match           node_exporter*
  allowlist_key   node_memory*
  allowlist_key   node_cpu_seconds*
```

**Comment by patrick-stephens** [comment]
I guess an issue may be for streams with both and duplicate keys, e.g. OTEL output can support all of them so a processor attached to it needs to handle both. We should have a way to handle it but it may be filter specific as well.
If we could get a LUA filter to support separate callbacks for metrics/logs/etc. then that might be a good first priority to allow any behaviour.

**Comment by Spoutnik97** [comment]
I just found this issue, but I struggled with this behavior from 2 days...
I didn't see in the documentation that filters only works on json format. 
I tried to filter unused metrics from node_exporter and kube_state_metrics (scraped with prometheus)
